### PR TITLE
DynamicRoute Extension migration

### PR DESCRIPTION
**Pull request by Borales**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**
Might be (if somebody has already generated a sample app using `shop-with-blog` example - they need to remove `"package": "@deity/falcon-dynamic-route-extension"` ext config).

### CODE REVIEW

File: packages/falcon-server-env/src/types.ts [hunk 1]
```diff
@@ -8,6 +8,16 @@ export interface FetchUrlResult {
   type: string;
   path: string;
 }
+
+export enum ApiUrlPriority {
```
Comment (mateuszwozniak) [inline packages/falcon-server-env/src/types.ts]: Maybe we should revert that so the highest is indeed the highest, e.g.: 
```
  HIGHEST = 100,
  HIGH = 75,
  NORMAL = 50,
  LOW = 25,
  LOWEST = 1,
  OFF = 0
```

File: packages/falcon-server/docs/DynamicRouteResolver.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Dynamic Route Resolver
+
+`DynamicRouteResolver` handles dynamic route resolution for the client application.
+When user goes to an url that is not defined in React routing then client asks back-end for the type of content for that particular url:
+
+- It asks every extension to get "fetch URL priority" from the assigned API DataSource instance
+- It sorts API DataSource instances that are able to detect dynamic content type by their "priority"
+- It calls `fetchUrl` method o
```
Comment (mateuszwozniak) [inline packages/falcon-server/docs/DynamicRouteResolver.md]: Not finished?

File: packages/falcon-server/docs/DynamicRouteResolver.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    fetchUrl(obj, args, context, info) {
+        // implementation of the resolver
+    }
+}
+```
+
+```js
+class WordpressApoi {
```
Comment (mateuszwozniak) [inline packages/falcon-server/docs/DynamicRouteResolver.md]: `WordpressApi` :)

File: packages/falcon-server/docs/DynamicRouteResolver.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+So in that case `DynamicRouteResolver` will call `ShopExtension.fetchUrl()` first, if that call returns `null`,
+it will call `BlogExtension.fetchUrl()` second.
+
+## Requirements
+
+If any extension should handle dynamic routing it should implement both methods:
+
+- `getFetchUrlPriority(url)` - accepts url (String) that should be resolved and returns the priority (Number) for the extension
+- `fetchUrl(obj, args, context, info)` - fetches the data from the remote source or performs any other async logic to determine
+the type of the given url
```
Comment (mateuszwozniak) [inline packages/falcon-server/docs/DynamicRouteResolver.md]: Accidental new line?

File: packages/falcon-server-env/src/types.ts (hunk 1 above)
Comment (Borales) [reply packages/falcon-server-env/src/types.ts]: @mateuszwozniak changed

File: packages/falcon-server/docs/DynamicRouteResolver.md (hunk 2 above)
Comment (Borales) [reply packages/falcon-server/docs/DynamicRouteResolver.md]: @mateuszwozniak it is now

File: packages/falcon-server/docs/DynamicRouteResolver.md (hunk 3 above)
Comment (Borales) [reply packages/falcon-server/docs/DynamicRouteResolver.md]: @mateuszwozniak fixed

File: packages/falcon-server/docs/DynamicRouteResolver.md (hunk 4 above)
Comment (Borales) [reply packages/falcon-server/docs/DynamicRouteResolver.md]: @mateuszwozniak fixed

**Pull request merged**